activedirectory.sgnl.yaml

```diff
@@ -0,0 +1,229 @@
+description: "Active Directory as a System of Record"
+hostname: "{{Input Required: Hostname}}"
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "ActiveDirectory-1.0.0"
+# Example Config:
+#{
+#    "baseDN":"dc=org,dc=example,dc=io",
+#    "certificateChain": ". . . base64 encoded certificate chain here . . ",
+#    "filters":{
+#         "User":"(&(objectClass=user)(objectCategory=user)(distinguishedName=*))",
+#         "Group":"(&(objectClass=group)(objectCategory=group)(distinguishedName=*))"
+#    }
+#}
+adapterConfig: "ewogICAgImJhc2VETiI6ImRjPW9yZyxkYz1leGFtcGxlLGRjPWlvIiwKICAgICJjZXJ0aWZpY2F0ZUNoYWluIjogIi4gLiAuIGJhc2U2NCBlbmNvZGVkIGNlcnRpZmljYXRlIGNoYWluIGhlcmUgLiAuICIsCiAgICAiZmlsdGVycyI6ewogICAgICAgICJVc2VyIjoiKCYob2JqZWN0Q2xhc3M9dXNlcikobG9nb25Db3VudD0wKSkiCiAgICB9Cn0="
+# All auth mechanisms specified below must be supported by the specified Adapter.
+auth:
+  - basic:
+      username: "{{Input Required: Username}}"
+      password: "{{Input Required: Password}}"
+entities:
+  User:
+    displayName: User
+    externalId: User
+    description: User Entity in Active Directory
+    pageSize: 999
+    pagesOrderedById: false
+    attributes:
+      - name: dn
+        externalId: dn
+        description: distinguished Name of the user
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: obejctClass
+        externalId: obejctClass
+        type: String
+      - name: sn
+        externalId: sn
+        type: String
+      - name: givenName
+        externalId: givenName
+        type: String
+      - name: distinguishedName
+        externalId: distinguishedName
+        type: String
```

Comment:
Maybe index this

Reply:
`dn` is correct indexed attribute. A distinguished name (usually just shortened to “DN”) uniquely identifies an entry and describes its position in the DIT. We are using `dn` as `uniqueIDAttribute` for Entity in the adapter.

Ref: https://ldap.com/ldap-dns-and-rdns/